src/codeu/chat/common/UserType.java

```diff
@@ -0,0 +1,49 @@
+package codeu.chat.common;
+
+import codeu.chat.util.Serializer;
+import codeu.chat.util.Serializers;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public enum UserType {
+  CREATOR(0),
+  OWNER(1),
+  MEMBER(2),
+  NOTSET(3);
+
+  public final int fId;
+
+  UserType(int id) {
+    fId = id;
+  }
+
+  public static final Serializer<UserType> SERIALIZER =
+      new Serializer<UserType>() {
+        @Override
+        public void write(OutputStream out, UserType value) throws IOException {
+          Serializers.INTEGER.write(out, value.fId);
+        }
+
+        @Override
+        public UserType read(InputStream in) throws IOException {
+          return UserType.fromId(Serializers.INTEGER.read(in));
+        }
+      };
+
+  public static UserType fromId(int id) {
+    return values()[id];
```

Comment:
Can you handle `ArrayOutOfBoundsException` with a more useful error like `IllegalArgumentException`? 

I don't like exposing the underlying `int`. If its for serialization, please add a comment warning against other uses.